src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -1062,7 +1090,11 @@ std::vector<event::ptr> loop_inst::handle_buffers_for_next_iteration(const loop_
             if (mapping.from_mem == nullptr) {
                 mapping.from_mem = mapping.from_primitive->output_memory_ptr();
                 OPENVINO_ASSERT(mapping.from_mem != nullptr, "from_mem should not be null");
-                set_memory_in_body_network(body_network, mapping.to_primitive, mapping.from_mem);
+                if (mapping.to_primitive->get_output_layout(0).count() < mapping.from_primitive->get_output_layout(0).count()) {
+                    set_memory_in_body_network_from_output(body_network, mapping.to_primitive, mapping.from_mem);
```

Comment:
Can we create test cases to verify this behavior?

Reply:
Hi River
We already have the case TF_Faster_RCNN_Inception_ResNet_v2_atrous_coco that includes the loop which will utilize this behavior. is that case enough or we need to create more cases like that?

Reply:
It is `loop` behavior, it will be better to create dedicated case to test it.

Reply:
got it, I'll try to create one.